charms/trusty/clearwater-aio-proxy/lib/reconfigure-aio

```diff
@@ -15,19 +15,27 @@ fi
 old_home_domain=$(. /etc/clearwater/config ; echo $home_domain)
 echo "DELETE FROM ellis.numbers WHERE number LIKE '%@$old_home_domain' AND owner_id IS NULL ;" | mysql
 
-# Update /etc/clearwater/shared_config.
-function escape { echo $1 | sed -e 's/\//\\\//g' ; }
-sed -e 's/^home_domain=.*$/home_domain='$(escape $home_domain)'/g' \
-    </etc/clearwater/shared_config >/tmp/shared_config.$$
-mv /tmp/shared_config.$$ /etc/clearwater/shared_config
-
-# Restart clearwater-infrastructure to propagate changes to other configuration files.
-service clearwater-infrastructure restart
+# Update /etc/clearwater/shared_config, if the home domain has changed.
+if [ "$home_domain" != "$old_home_domain" ] ; then
+  function escape { echo $1 | sed -e 's/\//\\\//g' ; }
```

Comment:
You could define the function outside the if (and that feels more natural) but if you've already tested this I don't feel strongly.